ndcube/ndcube.py

```diff
@@ -321,111 +346,69 @@ def extra_coords(self):
-                             for i in range(n_dim)]
-
-        # Convert them back to units of world_axis_units
-        world_axis_units = self.wcs.low_level_wcs.world_axis_units[::-1]
-        all_world_corners = [entries.to(world_axis_units[i]) for i, entries in enumerate(all_world_corners)]
-
-        # Here we are using `wcs.world_to_pixel_values` instead of NDCube's world_to_pixel as the latter
-        # requires high_level astropy objects to operate
-        # Since it is a low_level API, so input parameters need to be adjusted in wcs ordering.
-
-        # Convert to pixel coordinates
-        all_world_corners = all_world_corners[::-1]
-        all_pix_corners = self.wcs.low_level_wcs.world_to_pixel_values(*all_world_corners)
-        all_pix_corners = all_pix_corners[::-1]
-        all_pix_corners = tuple(u.Quantity(a, u.pix).value for a in all_pix_corners)
-
-        # Derive slicing item with which to slice NDCube.
-        # Be sure to round down min pixel and round up + 1 the max pixel.
-        item = tuple([slice(int(np.clip(axis_pixels.min(), 0, None)),
-                            int(np.ceil(axis_pixels.max())) + 1)
-                      for axis_pixels in all_pix_corners])
-        return self[item]
-
-    def crop_by_extra_coord(self, coord_name, min_coord_value, max_coord_value):
-        """
-        Crops an NDCube given a minimum value and interval width along an extra
-        coord.
-
-        Parameters
-        ----------
-        coord_name: `str`
-            Name of extra coordinate by which to crop.
-
-        min_coord_value: Single value `astropy.units.Quantity`
-            The minimum desired value of the extra coord after cropping.
-            Unit must be consistent with the extra coord on which cropping is based.
-
-        min_coord_value: Single value `astropy.units.Quantity`
-            The maximum desired value of the extra coord after cropping.
-            Unit must be consistent with the extra coord on which cropping is based.
-
-        Returns
-        -------
-        result: `ndcube.NDCube`
-        """
-        if not isinstance(coord_name, str):
-            raise TypeError("The API for this function has changed. "
-                            "Please give coord_name, min_coord_value, max_coord_value")
-        extra_coord_dict = self.extra_coords[coord_name]
-        if isinstance(extra_coord_dict["value"], u.Quantity):
-            extra_coord_values = extra_coord_dict["value"]
+    def crop(self, *intervals, wcs=None):
+        # The docstring is defined in NDCubeBase
+        return self._crop(*intervals, wcs=wcs, crop_by_values=False)
+
+    def crop_by_values(self, *intervals, wcs=None):
+        # The docstring is defined in NDCubeBase
+        return self._crop(*intervals, wcs=wcs, crop_by_values=True)
+
+    def _crop(self, *intervals, wcs=None, crop_by_values=False):
+        # If no intervals provided, return NDCube without slicing.
```

Comment:
would an error not be more appropriate here?

Reply:
This comment should read `# If all intervals provides are None, return NDCube without slicing.`  I see this as the equivalent to `same_cube = cube[:, ..., :]`

Reply:
Is that a use that we really need to allow here lol? I don't really understand why this would be useful.

Reply:
Supporting it doesn't change the code in any way except that an error is or isn't raise in line 361.  Checking whether all intervals are `None` is still required.  And if regular slicing allows all axes to be sliced by `slice(None)`, we might as well mimic that behaviour for `crop`.